From 91b4fdefc5836c259b783e56d77ff3e27ad62236 Mon Sep 17 00:00:00 2001
From: Tomas Hozza <thozza@redhat.com>
Date: Fri, 27 May 2016 10:21:15 +0200
Subject: [PATCH] hashsize() return type changed in libdns v164

Since bind version 9.10.4b1 (libdns version 164), the return type of
hashsize() has changed from unsigned int to size_t. Without this change
the plugin does not compile against bind 9.10.4b1 or newer on 64bit
architecture.

Signed-off-by: Tomas Hozza <thozza@redhat.com>
---
 src/ldap_driver.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ldap_driver.c b/src/ldap_driver.c
index 5727641..83ec00a 100644
--- a/src/ldap_driver.c
+++ b/src/ldap_driver.c
@@ -871,7 +871,11 @@ setcachestats(dns_db_t *db, isc_stats_t *stats)
 	return dns_db_setcachestats(ldapdb->rbtdb, stats);
 }
 
+#if LIBDNS_VERSION_MAJOR >= 164
+size_t
+#else
 unsigned int
+#endif /* LIBDNS_VERSION_MAJOR >= 164 */
 hashsize(dns_db_t *db)
 {
 	ldapdb_t *ldapdb = (ldapdb_t *) db;
-- 
2.5.5

